File: Python/day1/day1.py
def part1(input):
    temp = []
    inputs = []
    for inp in input:
        if inp == '':
            inputs.append(sum(temp))
            temp = []
            pass
        else:
            temp.append(int(inp))
    inputs.append(temp[0])
    return max(inputs)

def part2(input):
    temp = []
    inputs = []
    for inp in input:
        if inp == '':
            inputs.append(sum(temp))
            temp = []
        else:
            temp.append(int(inp))
    inputs.append(temp[0])
    inputs.sort()
    inputs.reverse()
    return sum(inputs[0:3])


        



if __name__ == "__main__":
    train = [line.replace('\n', '') for line in open("day1/inputs/train.txt", "r")]
    test = [line.replace('\n', '') for line in open("day1/inputs/test.txt", "r")]

    assert part1(train) == 24000
    assert part1(test) == 66616
    assert part2(train) == 45000
    assert part2(test) == 199172
    
    print(f"Train Part 1: {part1(train)}")
    print(f"Test Part 1: {part1(test)}")
    print(f"Train Part 2: {part2(train)}")
    print(f"Test Part 1: {part2(test)}")

File: elixiraoc/lib/day1.ex
defmodule Day1 do
  def test do
    IO.puts("Train part1: #{part1("inputs/day1/train.txt")}")
    IO.puts("Test part1: #{part1("inputs/day1/test.txt")}")
    IO.puts("Train part2: #{part2("inputs/day1/train.txt")}")
    IO.puts("Test part2: #{part2("inputs/day1/test.txt")}")
  end

  def part1(path) do
    input(path)
    |> Enum.chunk_by(&(&1 == ""))
    |> Enum.filter(&(&1 != [""]))
    |> Enum.map(&(Enum.map(&1, fn x -> String.to_integer(x) end)))
    |> Enum.map(&(Enum.sum(&1)))
    |> Enum.max()
  end

  def part2(path) do
    input(path)
    |> Enum.chunk_by(&(&1 == ""))
    |> Enum.filter(&(&1 != [""]))
    |> Enum.map(&(Enum.map(&1, fn x -> String.to_integer(x) end)))
    |> Enum.map(&(Enum.sum(&1)))
    |> Enum.sort(:desc)
    |> Enum.take(3)
    |> Enum.sum()
  end

  def input(path), do: path |> File.stream!() |> Enum.map(&parse_input/1)

  def parse_input(input), do: input |> String.trim()
end
